### PR TITLE
Fix yearn instant rebind price

### PR DESCRIPTION
**Pull request by defi-dev**

### CODE REVIEW

**Review by defi-lead — APPROVED** [review]

File: contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol [hunk 1]
```diff
@@ -522,53 +527,41 @@ contract YearnVaultInstantRebindStrategy is SinglePoolManagement, WeightValueCha
...
     oldBalances = new uint256[](len);
-    poolUSDCBalances = new uint256[](len);
-    totalUSDCPool = USDC.balanceOf(address(this));
+    vaultUSDCPrices = new uint256[](len);
+    totalUSDCPool = USDC.balanceOf(address(this)) * 1e12;
 
     for (uint256 oi = 0; oi < len; oi++) {
+      uint256 vaultUSDCPrice =
+        bdiv(
+          getVaultVirtualPriceEstimation(_tokens[oi], IYearnVaultV2(_tokens[oi]).totalAssets()),
```
Comment (eMarchenko) [inline contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol:554]: this line is effectively  `getTVL(_tokens[oi])`, isn't it?

File: contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol [hunk 2]
```diff
@@ -426,7 +431,7 @@ contract YearnVaultInstantRebindStrategy is SinglePoolManagement, WeightValueCha
         (mem.ycrvBalance, mem.crvActual, mem.usdcBefore) = _vaultToUsdc(cfg.token, mem.crvToken, vc);
 
         // 2nd step. Vault.withdraw()
-        mem.crvExpected = (mem.ycrvBalance * IYearnVaultV2(cfg.token).pricePerShare()) / 1e18;
+        mem.crvExpected = bmul(mem.ycrvBalance, IYearnVaultV2(cfg.token).pricePerShare());
 
         emit PullLiquidity(
           cfg.token,
```
Comment (eMarchenko) [inline contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol:437]: bmul at line 447

File: contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol [hunk 3]
```diff
@@ -426,7 +431,7 @@ contract YearnVaultInstantRebindStrategy is SinglePoolManagement, WeightValueCha
         (mem.ycrvBalance, mem.crvActual, mem.usdcBefore) = _vaultToUsdc(cfg.token, mem.crvToken, vc);
```
Comment (eMarchenko) [inline contracts/weight-strategies/YearnVaultInstantRebindStrategy.sol:431]: `mem.usdcBefore` is calculated inside `_vaultToUsdc` function as  `usdcBefore = USDC.balanceOf(address(this));` and the value is not used inside `_vaultToUsdc`.
Consider calculating "usdcBefore" inside `_instantRebind` - it should be in the same function as "usdcAfter" (calculated at L442)

**Review by eMarchenko — COMMENTED** [review]
minor code quality issues that don't directly affect security

**Pull request merged**